libs/decoupling/brain/src/lib/button-disabled.directive.ts

```diff
@@ -1,20 +1,31 @@
 /* eslint-disable @angular-eslint/directive-selector */
 /* eslint-disable @angular-eslint/no-host-metadata-property */
+import {
+  BUTTON_STATE,
+  ButtonSignalState,
+  ButtonState,
+} from '@angular-challenges/decoupling/core';
 import { Directive, WritableSignal, signal } from '@angular/core';
 
-export type ButtonState = 'enabled' | 'disabled';
-
 @Directive({
   selector: 'button[btnDisabled]',
   standalone: true,
   host: {
     '(click)': 'toggleState()',
   },
+  providers: [
+    {
+      provide: BUTTON_STATE,
+      useExisting: BtnDisabledDirective,
```

Comment:
There is no error ? I thought you need forwardRef since BtnDisableDirective is not yet instanciated ?

Reply:
I wasn't getting an error. I saw that you used `forwardRef` after I did mine, but I couldn't see a difference in the app.

Reply:
Ok that's weird.😅 I will retry this next week.

Reply:
I don't understand `forwardRef` that well, so I could definitely be missing something.

Reply:
Yes you are right, it's working without forwardRef. I don't get why 🤔